fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -777,8 +795,25 @@ public void replaceProperties(final IdentifierConverter<Resource, FedoraResource
                 exceptions.append("\n");
             }
 
-            try (final DefaultRdfStream notCommonStream =
-                    new DefaultRdfStream(replacementStream.topic(), differencer.notCommon())) {
+            try (
+                final DefaultRdfStream notCommonStream =
+                        new DefaultRdfStream(replacementStream.topic(), differencer.notCommon());
+                final DefaultRdfStream testStream =
+                        new DefaultRdfStream(replacementStream.topic(), differencer.notCommon())) {
+
+                // do some very basic validation to catch invalid RDF
+                // this uses the same checks that updateProperties() uses
+                final Collection<IllegalArgumentException> errors = testStream
+                        .map(x -> Quad.create(x.getSubject(), x))
+                        .flatMap(FedoraResourceImpl::validateQuad)
+                        .filter(x -> x != null)
+                        .collect(Collectors.toList());
+
+                if (!errors.isEmpty()) {
+                    throw new ConstraintViolationException(
```

Comment:
I haven't tested this but, the `FedoraResourceImplIT` tests made me wonder, it seems like you might get a different error for the same problem depending on the operation. ConstrainViolationException or IllegalArgumentException... or is that just in the tests?

Reply:
Ok, so the end result is a `400 Bad Request` to either `PATCH`ing or `PUT`ing the `/fcr:metadata` endpoint or `PUT`ing with an invalid `Content-type` header. So you can disregard this.